opm/material/components/CO2Tables.hpp

```diff
@@ -57,16 +57,51 @@ struct co2TabulatedEnthalpyTraits {
     static const Scalar vals[200][500];
 };
 
+template<class Scalar = double, class ContainerT = std::vector<double>>
 class CO2Tables
 {
 public:
-    Opm::UniformTabulated2DFunction<double> tabulatedEnthalpy;
-    Opm::UniformTabulated2DFunction<double> tabulatedDensity;
+    Opm::UniformTabulated2DFunction<Scalar, ContainerT> tabulatedEnthalpy;
+    Opm::UniformTabulated2DFunction<Scalar, ContainerT> tabulatedDensity;
     static constexpr double brineSalinity = 1.000000000000000e-01;
 
     CO2Tables();
+
+    CO2Tables(const Opm::UniformTabulated2DFunction<Scalar, ContainerT>& enthalpy,
+              const Opm::UniformTabulated2DFunction<Scalar, ContainerT>& density)
+        : tabulatedEnthalpy(enthalpy), tabulatedDensity(density)
+    {
+    }
+
+    const Opm::UniformTabulated2DFunction<Scalar, ContainerT>& getTabulatedEnthalpy() const {
+        return tabulatedEnthalpy;
+    }
+
+    const Opm::UniformTabulated2DFunction<Scalar, ContainerT>& getTabulatedDensity() const {
+        return tabulatedDensity;
+    }
 };
 
 } // namespace Opm
 
+namespace Opm::gpuistl {
+    template <class Scalar, class ContainerType, class ViewType>
+    CO2Tables<Scalar, ViewType>
+    make_view(const CO2Tables<Scalar, ContainerType>& oldCO2Tables) {
```

Comment:
As elsewhere: makeView() instead of make_view() etc.

Reply:
On another note, unless there are external requirements that the template parameters be in the order

1. `Scalar`
2. `ContainerType`
3. `ViewType`

I think I'd normally reorder them a bit to have the `ViewType` first&ndash;e.g.,
```C++
template <typename ViewType, typename Scalar, typename ContainerType>
CO2Tables<Scalar, ViewType> makeView(const CO2Tables<Scalar, ContainerType>& inputTable)
{
    // ...
}
```
That way, the caller need only pass one explicit template argument while the rest would be inferred through normal template argument deduction, e.g.,
```C++
const auto table_view = makeView<ViewType>(inputTables);
```
whence `table_view` would be a `CO2Tables<Scalar, ViewType>` and the `Scalar` would be inferred from `inputTables`.

Reply:
make_view and move_to_gpu are already used other places for this purpose, i'd rather go back later and fix them all together so the code using these functions always has a consistent way of doing it